server/neptune/workflows/internal/deploy/revision/validator.go

```diff
@@ -0,0 +1,63 @@
+package revision
```

Comment:
PR slightly clashes with #381 
Is the intention here to lock for both unexpected revisions and force applies using this logic, or is this completely separate for non-force applies: https://github.com/lyft/atlantis/blob/8ea4ca096231c4229030b7117b837095cc668cc6/server/neptune/workflows/internal/terraform/root_locker.go#L54

Reply:
Since force applies are directly routed to the terraform workflow, do we need to support diverged commits in the deploy workflow? Ideally, we should only receive push events from the main branch right?  

I initially added a TODO comment to add force apply logic in here but since force apply request never gets here, I don't think we need to add anything here. I've removed the TODO comment. 

Sorry for the confusion 😄